cardano-config/src/Cardano/Config/Types.hs

```diff
@@ -163,9 +177,9 @@ newtype DelegationCertFile = DelegationCertFile
   { unDelegationCert :: FilePath }
   deriving Show
 
-newtype SocketFile = SocketFile
-  { unSocket :: FilePath }
-  deriving Show
+data SocketPath = SocketFile FilePath
+                | SocketDir FilePath
```

Comment:
Do we have a TODO or ticket to remove the dir case again once we finish eliminating the node ids?

Reply:
@dcoutts I was under the impression we still needed the `NodeId`s for the mock protocols and I believe DevOps finds it easier to work with socket dirs. @deepfire can you clarify re: socket dirs?

Reply:
The mock/test protocols need the node ids for the consensus algorithms. I'm pretty sure devops have said they want simple "do exactly what I tell you" style with files, and not impose any naming convention.

Reply:
I think @deepfire wanted the ability to specify a socket dir for scripts. If not I can remove it.

Reply:
It can be removed -- but then all the users will have to be updated.